src/Compilers/CSharp/Portable/Syntax/SyntaxNormalizer.cs

```diff
@@ -654,7 +654,9 @@ private static bool NeedsSeparator(SyntaxToken token, SyntaxToken next)
                 }
             }
 
-            if (token.IsKind(SyntaxKind.GreaterThanToken) && token.Parent.IsKind(SyntaxKind.FunctionPointerParameterList))
+            if (token.IsKind(SyntaxKind.GreaterThanToken) &&
+                token.Parent.IsKind(SyntaxKind.FunctionPointerParameterList) &&
+                token.Parent.Parent?.Parent is not UsingDirectiveSyntax)
```

Comment:
>token.Parent.Parent?.Parent [](http://example.com/codeflow?start=16&length=27)

Is it enough to look only two levels up. What if the function pointer is nested deeper. For example, into an array type or into a type argument?

Reply:
Then that won't matter as it won't be followed by the semicolon.  e.g. `delegate*<int,int>[];` will be a question about how `>[` is normalized (which is a present issue, unrelated to my pr), and how `];` is normalized (which is covered by another test).

Reply:
> Then that won't matter as it won't be followed by the semicolon.  

Where do we check for the fact? Specifically: "followed by the semicolon."

Reply:
implied by: `token.Parent.Parent?.Parent is not UsingDirectiveSyntax)`.  we know (from line above) that token.Parent is FuncPointerParameterList, to token.Parent.Parent is FuncPointerSYntax.  So if token.Parent.Parent.Parent *is* a UsingDirectiveSyntax, then the func-pointer showed up as the `NamespaceOrType` of that using directive.  So, in that case, we do not want a whitespace separator between it and the next token (which would be the semicolon of the using directive).

Reply:
> So, in that case, we do not want a whitespace separator between it and the next token (which would be the semicolon of the using directive)

That is understood. The question however is why the same logic should not be applied to a case when we are the "using" type, but not at the top level?

Reply:
In other words, are there more scenarios that we are missing?

Reply:
those scenarios would also arise for function-pointer *prior* to my PR/feature.  So i view it as out of the scope of what this PR is trying to accomplish.  IN other words, say there is an issue with an array of function pointers, where you get it as `delegate*<int, int> []` instead of `delegate*<int, int>[]`.  That issue would exist regardless of the type being in a using directive or not.  It's possible there is some issue, but it doesn't seem related/relevant to the feature i'm doing, so it's not something i'm looking into improving.